src/main/java/Duke.java

```diff
@@ -21,58 +24,95 @@ public static void runDuke() {
         greeting();
 
         Scanner sc = new Scanner(System.in);
-        String input;
-        Task[] list = new Task[100];
-        int numTasks = 0;
 
-        while (!(input = sc.nextLine()).equals("bye")) {
-            String command = input.split(" ")[0];
+        while (true) {
+            String input = sc.nextLine();
 
-            if (command.equals("list")) {
-                printList(list, numTasks);
-            } else if (command.equals("done")) {
-                int taskNum = Integer.parseInt(input.split(" ")[1]);
-                list[taskNum - 1].markAsDone();
-                System.out.println(delimiter());
-                System.out.println("I've finally done this task:");
-                System.out.println(list[taskNum - 1].toString());
-                System.out.println(delimiter() + "\n");
-            } else {
-                Task newTask;
-                if (command.equals("todo")) {
-                    newTask = new ToDo(getDescription(input));
-                } else if (command.equals("deadline")) {
-                    newTask = new Deadline(getDescription(input), getTime(input));
-                } else { // (command.equals("event")) {
-                    newTask = new Event(getDescription(input), getTime(input));
+            try {
+                String command = getCommand(input);
+                if (command.equals("bye")) {
+                    break;
                 }
+                doInstructions(input);
+            } catch (DukeException e) {
+                System.out.printf("Error: %s\n\n", e.getMessage());
+            }
+        }
+        System.out.println("Bye! Hope you don't come back!");
+    }
 
-                list[numTasks] = newTask;
-                numTasks++;
-                System.out.println(delimiter());
-                System.out.println("Alrighty, you added:");
-                System.out.println("  " + newTask.toString());
-                System.out.printf("Now you have %d tasks in your list.\n", numTasks);
-                System.out.println(delimiter() + "\n");
+    public static void doInstructions(String input) throws DukeException {
+        Task[] list = new Task[100];
+        int numTasks = 0;
+        String command = getCommand(input);
+
+        if (command.equals("list")) {
+            printList(list, numTasks);
+        } else if (command.equals("done")) {
+            int taskNum = Integer.parseInt(input.split(" ")[1]);
+            // check for
+            // 1. invalid task num
+            // 2. empty task num
+            // 3. if task is already done?
+            // implement ui class? if have time to print out done statement
+            list[taskNum - 1].markAsDone();
+            System.out.println(delimiter());
+            System.out.println("I've finally done this task:");
+            System.out.println(list[taskNum - 1].toString());
+            System.out.println(delimiter() + "\n");
```

Comment:
Could potentially combine the println statements into a single command, separating with + and \n where needed.